File: py_payment_api/app/services/store/payment/one_time/one_time_payment_item_service.py
from ..abstract.item_collection_service import ItemCollectionService
from enum import StrEnum
from typing import TypeVar, Generic
from ....store.enum import ItemType
ITEM_CATEGORY = TypeVar('ITEM_CATEGORY', bound=StrEnum)

class OneTimePaymentItemService(Generic[ITEM_CATEGORY], ItemCollectionService[ITEM_CATEGORY]):
    def __init__(self) -> None:
        super().__init__(ItemType.ONE_TIME_PAYMENT, ItemType.ONE_TIME_PAYMENT)

File: tsPaymentApi/src/services/store/payment/one_time/OneTimePaymentItemService.ts
import { ItemCollectionService } from '../abstract/ItemCollectionService';
import { ItemType } from '../../../../types/store';
import { AvailableItem } from '../../../../types/store/AvailableItem';

export class OneTimePaymentItemService<TCategory extends string> extends ItemCollectionService<TCategory> {
    constructor() {
        super(ItemType.ONE_TIME_PAYMENT, ItemType.ONE_TIME_PAYMENT);
    }

    async getItems(): Promise<AvailableItem[]> {
        // Implementation to be added
        return [];
    }

    async getItemByName(name: string): Promise<AvailableItem | null> {
        // Implementation to be added
        return null;
    }

    async validateItem(name: string): Promise<boolean> {
        // Implementation to be added
        return false;
    }
} 